File: internal/config/session_secret_gate.go
package config

import "errors"

// defaultSessionSecret ist das im Repo oeffentlich stehende Platzhalter-Secret.
const defaultSessionSecret = "dev-secret-change-me"

// minSessionSecretLen ist die Mindestlaenge fuer ein individuelles Secret.
const minSessionSecretLen = 32

// ValidateSessionSecret prueft, ob das Session-Secret fuer den Produktivbetrieb
// taugt.
//
// Issue #2139: Mit dem unveraenderten Default-Secret kann jede Person ein
// gueltiges Session-Cookie fuer eine beliebige user_id selbst signieren. Der
// Aufrufer (cmd/server/main.go) beendet den Prozess beim Start, statt unsicher
// weiterzulaufen.
//
// Ausnahme: Ist TestFixtureDir gesetzt, laeuft der Prozess im CI-E2E-Stack
// (frontend/e2e/ci-stack.sh) bewusst ohne GZ_SESSION_SECRET.
func ValidateSessionSecret(cfg *Config) error {
	if cfg.TestFixtureDir != "" {
		return nil
	}
	if cfg.SessionSecret == "" {
		return errors.New("GZ_SESSION_SECRET ist nicht gesetzt")
	}
	if cfg.SessionSecret == defaultSessionSecret {
		return errors.New("GZ_SESSION_SECRET traegt noch das Default-Literal")
	}
	if len(cfg.SessionSecret) < minSessionSecretLen {
		return errors.New("GZ_SESSION_SECRET ist kuerzer als 32 Zeichen")
	}
	return nil
}

File: frontend/src/lib/sessionSecretGate.ts
// Issue #2139: Fail-Fast fuer das Session-Secret im Frontend-Server.
//
// Ohne diese Pruefung fiele hooks.server.ts still auf das oeffentlich im Repo
// stehende Default-Secret zurueck — damit koennte jede Person ein gueltiges
// gz_session-Cookie fuer eine beliebige user_id selbst signieren.

const DEFAULT_SESSION_SECRET = 'dev-secret-change-me';
const MIN_SESSION_SECRET_LEN = 32;

export function assertSessionSecretConfigured(secret: string | undefined): void {
	if (!secret) {
		throw new Error('GZ_SESSION_SECRET ist nicht gesetzt');
	}
	if (secret === DEFAULT_SESSION_SECRET) {
		throw new Error('GZ_SESSION_SECRET traegt noch das Default-Literal');
	}
	if (secret.length < MIN_SESSION_SECRET_LEN) {
		throw new Error('GZ_SESSION_SECRET ist kuerzer als 32 Zeichen');
	}
}
